src/mapping/mapper.js

```diff
@@ -0,0 +1,184 @@
+// mappers to be executed they the filter would have to match for the containing
+// mapper.  This class can contain other aggregate mappers.
+class AggregateMapper {
+
+  constructor(args){
+    this.args = args;
+    this.filterFn = buildFilter(args.filter);
+    this.defaultFn = buildProcessor(args.default);
+    this.ignoreFn = buildFilter(args.ignore);
+    this.excludeFn = buildFilter(args.exclude);
+    this.mappers = buildMappers(args.mappers);
+  }
+
+  // if an ignore filter was provided execute it on the resource otherwise
+  // return false
+  ignore(resource){
+    return this.ignoreFn ? this.ignoreFn(resource) : false
+  }
+
+  // if an exclude filter was provided execute it on the resource otherwise return false
+  exclude(resource){
+    return this.excludeFn ? this.excludeFn(resource) : false
+  }
+
+  // if a default function was provided execute that function on the resource otherwise
+  // return the resource as is
+  default(resource){
+    return this.defaultFn ? this.defaultFn(resource) : resource;
+  }
+
+  // if a filter was provided execute that on the resource otherwise
+  // return false
+  filter(resource){
+    return (this.filterFn) ? this.filterFn(resource) : false;
+  }
+
+  // This method executes the aggregate filters.  There is a set order of operations
+  // for this method on a resource or set of resources passed in.
+  // ignore the resource if it returns true from the ignore function or does not pass the filter
+  // return null if the resource matches the exclude method
+  // if the resource matches a mapper that this aggregate mapper contains apply that mapper
+  // if the resource does not match a contained mapper run the default function on the resource
+  //
+  execute(resource){
+    if(Array.isArray(resource)){
+      return resource.map( r => this.execute(r)).filter(n => n)
+    }else{
+      if(this.ignore(resource) || !this.filter(resource)){return resource};
+      if(this.exclude(resource)){return null;};
+      let mapper = this.mappers.find(map => map.filter(resource));
+      if(mapper){
+        return mapper.execute(resource);
+      }else{
+        return this.default(resource);
+      }
+    }
+  }
+}
+
+// Mapper that does the actual work of modifying a reasource.  These are the leaf
+// nodes of aggregate mappers.  The class contains a filter that must be matched by the
+// aggregate mapper and an exec function that will modify the resource.
+class FilterMapper {
+
+  constructor(args){
+    this.args = args;
+    this.filterFn = buildFilter(args.filter);
+    this.execfn = buildProcessor(args.exec);
+  }
+
+  // if a filter was provided execute that function on the resource otherwise
+  // return false
+  filter(resource){
+    return (this.filterFn) ? this.filterFn(resource) : false;
+  }
+
+  execute(resource){
+    if(Array.isArray(resource)){
+      return resource.map( r => this.execute(r)).filter(n => n)
+    }
+    return this.execfn(resource);
+  }
+}
+
+module.exports = {
```

Comment:
Should the exports include the `buildMappers` function? That seems like it would be the primary entry point

Reply:
Looks like I forgot to push one of the updates which had the liniting errors and exports covered.  Just pushed